File: Workshops/PlantIdentifier/app/src/main/java/com/example/plantidentifier/MainActivity.java
package com.example.plantidentifier;

import androidx.activity.result.ActivityResult;
import androidx.activity.result.ActivityResultCallback;
import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.graphics.Bitmap;
import android.net.Uri;
import android.os.Bundle;
import android.provider.MediaStore;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.plantidentifier.ml.PlantsModel;

import org.tensorflow.lite.support.image.TensorImage;
import org.tensorflow.lite.support.label.Category;

import java.io.IOException;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    Button btLoadImage, btCaptureImage;
    TextView tvResult;
    ImageView ivAddImage;
   // ActivityResultLauncher<Intent> activityResultLauncher;
    ActivityResultLauncher<String> mGetContent;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        ivAddImage = findViewById(R.id.iv_add_image);
        tvResult = findViewById(R.id.tv_result);
        btLoadImage = findViewById(R.id.bt_load_image);
        // btCaptureImage = findViewById(R.id.bt_capture_image); to be implemented later live image capture

        //Picking image from gallery
        mGetContent = registerForActivityResult(new ActivityResultContracts.GetContent(), new ActivityResultCallback<Uri>() {
            @Override
            public void onActivityResult(Uri result) {
                Bitmap imageBitmap = null;
                try {
                    imageBitmap = UriToBitmap(result);
                } catch (IOException e) {
                    e.printStackTrace();
                }
                ivAddImage.setImageBitmap(imageBitmap);
                outputGenerator(imageBitmap);
                Log.d("TAG_URI", "" + result);
            }
        });

        btLoadImage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                mGetContent.launch("image/*");
            }
        });

        tvResult.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("https://www.google.com/search?q=" + tvResult.getText().toString()));
                startActivity(intent);
            }
        });
    }

    private void outputGenerator(Bitmap imageBitmap){
        try {
            PlantsModel model = PlantsModel.newInstance(MainActivity.this);

            // Creates inputs for reference.
            TensorImage image = TensorImage.fromBitmap(imageBitmap);

            // Runs model inference and gets result.
            PlantsModel.Outputs outputs = model.process(image);
            List<Category> probability = outputs.getProbabilityAsCategoryList();

            int index = 0;
            float max = probability.get(0).getScore();

            for(int i=0; i<probability.size(); i++){
                if(max<probability.get(i).getScore()){
                    max = probability.get(i).getScore();
                    index = i;
                }
            }

            Category output = probability.get(index);
            tvResult.setText(output.getLabel());

            // Releases model resources if no longer used.
            model.close();
        } catch (IOException e) {
            // TODO Handle the exception
        }

    }

    private Bitmap UriToBitmap(Uri result) throws IOException {
        return MediaStore.Images.Media.getBitmap(this.getContentResolver(), result);
    }
}

File: Individual Project/Vegiapp/app/src/main/java/com/example/vegia/MainActivity.kt
package com.example.vegia

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import org.w3c.dom.Text

class MainActivity : AppCompatActivity() {
    //getResources().getStringArray(R.array.your_array)[position]
    var x = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val challengeNumber = findViewById<TextView>(R.id.challengeTitle_txt)
        val challengeText = findViewById<TextView>(R.id.challenge_txt)
        val challengeList = resources.getStringArray(R.array.challenges)

        val sharedPreference =  getSharedPreferences("challengeNumber", Context.MODE_PRIVATE)
        if (sharedPreference.getInt("number",x) == null )
            x = 1
        else
            x = sharedPreference.getInt("number",x)


        challengeNumber.text = "Challenge "+x+":"
        challengeText.text = challengeList[x-1]

        val buttonClick = findViewById<Button>(R.id.complete_btn)
        buttonClick.setOnClickListener {
            val intent = Intent(this, ReceiveGem::class.java)
            startActivity(intent)
        }
    }
}